### PR TITLE
Bluetooth: Controller: Use HAL_TICKER_US_TO_TICKS_CEIL for ticks_slot

### PR DESCRIPTION
**Pull request by cvinayak**
Use HAL_TICKER_US_TO_TICKS_CEIL for ticks_slot calculations.

### CODE REVIEW

File: tests/bsim/bluetooth/mesh/_mesh_test.sh [hunk 1]
```diff
@@ -68,8 +70,9 @@ function RunTest(){
     Execute \
       ${exe_name} \
       -v=${verbosity_level} -s=$s_id -d=$idx -sync_preboot -RealEncryption=1 \
-      -testid=$testid ${testids["${testid}"]} ${test_options}
+      -rs=$seed -testid=$testid ${testids["${testid}"]} ${test_options}
     let idx=idx+1
+    let seed=seed+1
```
Comment (PavelVPV) [inline tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: Could you elaborate why this change is needed? Will mesh tests work without it?

Comment (cvinayak) [reply tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: pb_remote_reprovision.sh   was failing in this PR.. Now I see that other tests are failing when devices have different random seed values.

Debuging with @m-alperen-sener, we were able to pass  pb_remote_reprovision.sh by having a small (2ms) delay in relay_send.

Instead of adding delay, we decided to have random seed value in test script

Comment (PavelVPV) [reply tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: We used to add/increase delays in a particular test in case when the issue is not caused by invalid implementation (for example: https://github.com/zephyrproject-rtos/zephyr/commit/e5fcca6e99a52334c0c726dfc1b0c432f8d944bb#diff-7df14fc51b85bcb032356d6314ea93067acb2b55ad69b8585edd8821709da43c). Changing the seed will affect other tests.

Comment (PavelVPV) [reply tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: I can have a look later today

Comment (PavelVPV) [reply tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: Please try this patch:
```
diff --git a/tests/bsim/bluetooth/mesh/src/test_provision.c b/tests/bsim/bluetooth/mesh/src/test_provision.c
index 171f8cccbb..36e3c1250a 100644
--- a/tests/bsim/bluetooth/mesh/src/test_provision.c
+++ b/tests/bsim/bluetooth/mesh/src/test_provision.c
@@ -554,26 +554,26 @@ static void node_configure_and_reset(void)
                                          &status));
        ASSERT_EQUAL(0, status);
 
-       k_sleep(K_SECONDS(1));
+       k_sleep(K_SECONDS(2));
 
        ASSERT_OK(bt_mesh_cfg_cli_mod_app_bind(0, current_dev_addr, current_dev_addr, 0x0,
                                           BT_MESH_MODEL_ID_HEALTH_SRV, &status));
        ASSERT_EQUAL(0, status);
 
-       k_sleep(K_SECONDS(1));
+       k_sleep(K_SECONDS(2));
 
        ASSERT_OK(bt_mesh_cfg_cli_mod_sub_add(0, current_dev_addr, current_dev_addr, 0xc000,
                                          BT_MESH_MODEL_ID_HEALTH_SRV, &status));
        ASSERT_EQUAL(0, status);
 
-       k_sleep(K_SECONDS(1));
+       k_sleep(K_SECONDS(2));
 
        ASSERT_OK(bt_mesh_cfg_cli_mod_pub_set(0, current_dev_addr, current_dev_addr,
                                          BT_MESH_MODEL_ID_HEALTH_SRV, &healthpub,
                                          &status));
        ASSERT_EQUAL(0, status);
 
-       k_sleep(K_SECONDS(1));
+       k_sleep(K_SECONDS(2));
 
        ASSERT_OK(bt_mesh_cfg_cli_node_reset(0, current_dev_addr, (bool *)&status));
```

Comment (cvinayak) [reply tests/bsim/bluetooth/mesh/_mesh_test.sh:75]: thanks will try it and get back on Monday

**Review by PavelVPV — DISMISSED** [review]

**Comment by carlescufi** [comment]
@mtpr-ot @wopu-ot could you please review?

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h [hunk 2]
```diff
@@ -26,7 +26,19 @@
...
+
+/* Macro to translate microseconds to tick units.
+ * NOTE: This returns the ceil value.
+ */
+#define HAL_TICKER_US_TO_TICKS_CEIL(x) \
+	( \
+		((uint32_t)(((((uint64_t) (x) * 1000000000UL) + \
+			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
+			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
+		HAL_TICKER_CNTR_MASK \
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h:41]: Would this be the same as 
```c
ceiling_fraction(((uint64_t) (x) * 1000000000UL), HAL_TICKER_CNTR_CLK_UNIT_FS) & HAL_TICKER_CNTR_MASK
```
?

File: tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h [hunk 3]
```diff
@@ -3,38 +3,57 @@
...
+#define HAL_TICKER_CNTR_MASK 0x00FFFFFF
+
 /* Macro to translate microseconds to tick units.
  * NOTE: This returns the floor value.
  */
-#define HAL_TICKER_US_TO_TICKS(x)                                                                  \
-	(((uint32_t)(((uint64_t)(x) * 1000000000UL) / 30517578125UL)) & HAL_TICKER_CNTR_MASK)
+#define HAL_TICKER_US_TO_TICKS(x) \
+	( \
+		((uint32_t)(((uint64_t) (x) * 1000000000UL) / \
```
Comment (Thalley) [inline tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:27]: The multiplication of `1000000000UL` to done several places in the controller code. Suggest to introduce a macro for, similar to the `NSEC_PER_MSEC` and similar macros

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ticker.h [hunk 4]
```diff
@@ -35,6 +35,17 @@
...
 
+/* Macro to translate microseconds to tick units.
+ * NOTE: This returns the ceil value.
+ */
+#define HAL_TICKER_US_TO_TICKS_CEIL(x) \
+	( \
+		((uint32_t)(((((uint64_t) (x) * 1000000000UL) + \
+			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
+			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
+		HAL_TICKER_CNTR_MASK \
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ticker.h:46]: Ditto with using `ceiling_fraction`

File: tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h [hunk 5]
```diff
@@ -29,6 +29,16 @@
...
+/* Macro to translate microseconds to tick units.
+ * NOTE: This returns the ceil value.
+ */
+#define HAL_TICKER_US_TO_TICKS_CEIL(x) \
+	( \
+		((uint32_t)(((((uint64_t) (x) * 1000000000UL) + \
+			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
+			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
+		HAL_TICKER_CNTR_MASK \
+	)
```
Comment (Thalley) [inline tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:41]: Which of the value(s) are vendor specific, since we need several versions of this macro? 

Could we introduce a generic macro and simply have the vendor specific value as input?

File: tests/bsim/bluetooth/mesh/src/test_provision.c [hunk 6]
```diff
@@ -554,26 +554,26 @@ static void node_configure_and_reset(void)
 					  &status));
 	ASSERT_EQUAL(0, status);
 
-	k_sleep(K_SECONDS(1));
+	k_sleep(K_SECONDS(2));
```
Comment (Thalley) [inline tests/bsim/bluetooth/mesh/src/test_provision.c:557]: Does the changes to these test imply that scheduling is now "slower"? 

I think the commit message could use a bit more reasoning/elaboration.

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/ticker.h]: done

File: subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ticker.h (hunk 4 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/openisa/hal/RV32M1/ticker.h]: done

File: tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h (hunk 3 above)
Comment (cvinayak) [reply tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:27]: done

File: tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h (hunk 5 above)
Comment (cvinayak) [reply tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:41]: The macro calculation is vendor specific. 
For platforms where one tick is one microsecond, then `HAL_TICKER_US_TO_TICKS_CEIL` would just be `(x)`. Hence, all the HAL_TICKER_* is currently vendor specific.

File: tests/bsim/bluetooth/mesh/src/test_provision.c (hunk 6 above)
Comment (cvinayak) [reply tests/bsim/bluetooth/mesh/src/test_provision.c:557]: Updated the commit message, that the additional delay in timeout is introduced that so advertising retries are considered in verifying the test criteria.

File: tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h (hunk 5 above)
Comment (Thalley) [reply tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:41]: > For platforms where one tick is one microsecond, then HAL_TICKER_US_TO_TICKS_CEIL would just be (x). 

But in that could, couldn't we just do

```c
#define HAL_TICKER_US_TO_TICKS_CEIL(x, factor) \
	( \
		((uint32_t)(((((uint64_t) (x) * (factor)) + \
			      HAL_TICKER_CNTR_CLK_UNIT_FS - 1U)) / \
			    HAL_TICKER_CNTR_CLK_UNIT_FS)) & \
		HAL_TICKER_CNTR_MASK \
	)
```

Comment (cvinayak) [reply tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:41]: On nRF calculation is specific to have integer arithmetic when the clock is 32768 Hz, i.e. (1/32768)*1000000000 = 30517578125.

But say the clock is 16 MHz, then the calculation is simply ((x) * 16), it is cleaner to have separate hal/ticker.h file for different platforms with differing clock frequencies.

Comment (Thalley) [reply tests/bluetooth/controller/mock_ctrl/include/hal/ticker_vendor_hal.h:41]: Fair point - No reason to change in this PR anyways :)

**Review by Thalley — DISMISSED** [review]

**Review by PavelVPV — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by cvinayak** [comment]
@PavelVPV @Thalley Please re-review/approve, I had forgotten to fix `ticker_vendor_hal.h` in unit tests.

**Review by Thalley — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
